tasks/apply-mapping/apply-mapping.yaml

```diff
@@ -62,6 +62,42 @@ spec:
             exit 0
         fi
 
+        # Function to handle incrementer logic
+        increment_tag() {
+            local tag_template="$1"
+            local repo="$2"
+
+            # Use `skopeo list-tags` to fetch all tags from the repository
+            existing_tags=$(skopeo list-tags docker://"${repo}" | jq -r '.Tags[]')
+            
+            # Remove `{{ incrementer }}` placeholder to get the version prefix for regex pattern
+            # shellcheck disable=SC2001
+            version_prefix=$(echo "${tag_template}" | sed 's/{{ incrementer }}//g')
+            tag_pattern="^${version_prefix}[0-9]+$"  # Build regex pattern dynamically
+
+            # Extract the numeric part of existing tags and find the max increment
+            max_increment=$(echo "${existing_tags}" | grep -E "${tag_pattern}" | sed -E "s/${version_prefix}//" \
+            | sort -nr | head -n1)
+
+            # Set increment to 1 if no matching tags, otherwise increment max found by 1
+            if [ -z "${max_increment}" ]; then
+                increment=1
+            else
+                increment=$((max_increment + 1))
```

Comment:
I think you can just do
`increment=$((max_increment + 1))`
It appears if `$max_increment` is set to the empty string, it will set it to 1

Reply:
Thanks for the suggestion! I’ve updated it to the simplified version—it’s cleaner and works the same. I appreciate the tip!